sos/report/plugins/foreman_proxy.py

```diff
@@ -0,0 +1,123 @@
+            "/etc/{}*/conf/".format(self.apachepkg),
+            "/etc/{}*/conf.d/".format(self.apachepkg)
+        ])
+
+        self.add_cmd_output([
+            'foreman-maintain service status',
+            'ls -lanR /root/ssl-build',
+        ])
+
+        if self.get_option('proxyfeatures'):
+            _host_f = self.exec_cmd('hostname -f')['output']
+            _host_f = _host_f.strip()
+            _cmd = 'curl --cacert /etc/foreman-proxy/foreman_ssl_ca.pem ' \
+                   'https://%s:9090/features' % _host_f
+            self.add_cmd_output(_cmd, suggest_filename='smart_proxy_features',
+                                timeout=10)
+
+        # collect http[|s]_proxy env.variables
+        self.add_env_var(["http_proxy", "https_proxy"])
+
+    def postproc(self):
+        install_logs = "/var/log/foreman-installer/"
+        logreg = r"((foreman.*)?(\"::(foreman(.*?)|katello).*)?((::(.*)::.*" \
+                 r"(passw|cred|token|secret|key).*(\")?:)|(storepass )" \
+                 r"|(password =)))(.*)"
+        self.do_path_regex_sub(install_logs, logreg, r"\1 ********")
+        # need to do two passes here, debug output has different formatting
+        log_debug_reg = (r"(\s)+(Found key: (\"(foreman(.*?)|katello)"
+                         r"::(.*(token|secret|key|passw).*)\") value:) "
+                         r"(.*)")
+        self.do_path_regex_sub(install_logs, log_debug_reg, r"\1 \2 ********")
+        # also hide passwords in yet different formats
+        self.do_path_regex_sub(
+            install_logs,
+            r"(\.|_|-)password(=\'|=|\", \")(\w*)",
+            r"\1password\2********")
+        self.do_path_regex_sub(
+            "/var/log/foreman-installer/foreman-proxy*",
+            r"(\s*proxy_password\s=) (.*)",
+            r"\1 ********")
+        self.do_path_regex_sub(
+            "/var/log/foreman-maintain/foreman-maintain.log*",
+            r"(((passw|cred|token|secret)=)|(password ))(.*)",
+            r"\1********")
+        # all scrubbing applied to configs must be applied to installer logs
+        # as well, since logs contain diff of configs
+        self.do_path_regex_sub(
+            r"(/etc/foreman-(.*)((conf)(.*)?))|(%s)" % install_logs,
+            r"((\:|\s*)(passw|cred|token|secret|key).*(\:\s|=))(.*)",
+            r"\1********")
+        # yaml values should be alphanumeric
+        self.do_path_regex_sub(
+            r"(/etc/foreman-(.*)((yaml|yml)(.*)?))|(%s)" % install_logs,
+            r"((\:|\s*)(passw|cred|token|secret|key).*(\:\s|=))(.*)",
+            r'\1"********"')
+
+# Let the base Foreman class handle the string substitution of the apachepkg
+# attr so we can keep all log definitions centralized in the main class
+
+
+class RedHatForemanProxy(ForemanProxy, RedHatPlugin):
+
+    apachepkg = 'httpd'
+
+
+class DebianForemanProxy(ForemanProxy, DebianPlugin, UbuntuPlugin):
+
+    apachepkg = 'apache'
```

Comment:
Apache is not used in a non-Katello installation. Since Katello is not supported on Debian, this feels odd.

Reply:
What webserver does foreman on Debian without katello use, then? How to identify what webserver is used by any foreman deployment? We should update it accordingly...

Reply:
Foreman uses Apache, but Foreman Proxy is its own webserver. On a Katello deployment it just happens to also deploy a reverse proxy in Apache that is there for Pulp and a reverse proxy to Foreman itself.